aws/resource_aws_customer_gateway.go

```diff
@@ -30,7 +30,7 @@ func resourceAwsCustomerGateway() *schema.Resource {
 				Type:         schema.TypeInt,
 				Required:     true,
 				ForceNew:     true,
-				ValidateFunc: validation.IntBetween(1, 65534),
+				ValidateFunc: validation.IntBetween(0, 4294967295),
```

Comment:
Unfortunately this is an esoteric problem that isn't caught by CI (https://github.com/terraform-providers/terraform-provider-aws/issues/8988), but we'll need to do something different here to support 32-bit platforms:

```console
$ gox -os='linux darwin windows freebsd openbsd solaris' -arch='386 amd64 arm' -osarch='!darwin/arm !darwin/386' -ldflags '-s -w -X aws/version.ProviderVersion=99.99.99 -X aws/version.ProtocolVersion=4' -output 'results/{{.OS}}_{{.Arch}}/terraform-provider-aws_v99.99.99_x4' .
Number of parallel builds: 15

-->   solaris/amd64: github.com/terraform-providers/terraform-provider-aws
-->       linux/arm: github.com/terraform-providers/terraform-provider-aws
-->     freebsd/arm: github.com/terraform-providers/terraform-provider-aws
-->     freebsd/386: github.com/terraform-providers/terraform-provider-aws
-->     openbsd/386: github.com/terraform-providers/terraform-provider-aws
-->       linux/386: github.com/terraform-providers/terraform-provider-aws
-->     linux/amd64: github.com/terraform-providers/terraform-provider-aws
-->   windows/amd64: github.com/terraform-providers/terraform-provider-aws
-->    darwin/amd64: github.com/terraform-providers/terraform-provider-aws
-->   openbsd/amd64: github.com/terraform-providers/terraform-provider-aws
-->     windows/386: github.com/terraform-providers/terraform-provider-aws
-->   freebsd/amd64: github.com/terraform-providers/terraform-provider-aws

6 errors occurred:
--> openbsd/386 error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int

--> freebsd/arm error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int

--> linux/386 error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int

--> linux/arm error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int

--> windows/386 error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int

--> freebsd/386 error: exit status 2
Stderr: # github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int
```

You can check it much quicker without `gox` via:

```console
$ GOOS=linux GOARCH=386 go build
# github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_customer_gateway.go:33:44: constant 4294967295 overflows int
```

We can see how long the cross-compilation testing takes in GitHub Actions, although that was the original blocker for implementing it in TravisCI.

Reply:
Aah yes, this is similar to https://github.com/terraform-providers/terraform-provider-aws/pull/1888.
Is changing the attribute from `TypeInt` to `TypeString` a breaking change?

Reply:
In practice it seems not to be.